src/notifications/v2/notifications.grant.service.ts

```diff
@@ -0,0 +1,279 @@
+                grant.closing
+                    ? this.GRANT_CLOSING_TEMPLATE_ID
+                    : this.GRANT_OPENING_TEMPLATE_ID,
+                personalisation,
+                `${this.GRANT_CLOSING_TEMPLATE_ID}-${Date.toString()}`, //"reference"
+            );
+        }
+    };
+
+    private sendGrantUpdatedEmailsToBatch = async (
+        batchOfSubscriptions: Subscription[],
+        grantId: string,
+    ) => {
+        const userServiceSubEmailMap =
+            await this.notificationsHelper.getUserServiceEmailsBySubBatch(
+                batchOfSubscriptions
+                    .map((subscription) => subscription.user.sub)
+                    .filter((sub) => sub),
+            );
+
+        for (const subscription of batchOfSubscriptions as Subscription[]) {
+            const unsubscribeUrl =
+                await this.notificationsHelper.buildUnsubscribeUrl({
+                    subscriptionId: grantId,
+                    user: subscription.user,
+                });
+
+            const contentfulGrant = await this.contentfulService.fetchEntry(
+                grantId,
+            );
+
+            const personalisation = {
+                unsubscribeUrl,
+                'name of grant': contentfulGrant.fields.grantName as string,
+                'link to specific grant': `${this.HOST}/grants/${contentfulGrant.fields.label}`,
+            };
+
+            const email = extractEmailFromBatchResponse(
+                userServiceSubEmailMap,
+                subscription,
+            );
+
+            await this.emailService.send(
+                email ?? (await subscription.user.decryptEmail()),
+                this.GRANT_UPDATED_TEMPLATE_ID,
+                personalisation,
+                `${this.GRANT_UPDATED_TEMPLATE_ID}-${new Date().toISOString()}`,
+            );
+        }
+    };
+
+    processGrantUpdatedNotifications = async () => {
+        console.log('Running Process Grant Updated Notifications...');
+
+        const grantIds = await this.grantService.findAllUpdatedGrants();
+        for (const grantId of grantIds) {
+            const subscriptions =
+                await this.subscriptionService.findAllByContentGrantSubscriptionId(
+                    grantId,
+                );
+
+            const batchesCount =
+                this.notificationsHelper.getNumberOfBatchesOfNotifications(
+                    subscriptions.length,
+                );
+
+            for (let i = 0; i < batchesCount; i++) {
+                const batch = this.notificationsHelper.getBatchFromObjectArray(
+                    subscriptions,
+                    i,
+                    batchesCount,
+                );
+                await this.sendGrantUpdatedEmailsToBatch(
+                    batch as Subscription[],
+                    grantId,
+                );
+            }
+        }
+
+        await this.contentfulService.updateEntries(grantIds, {
```

Comment:
This function makes a lot of logs, would get rid of it logging the whole contentful entry and make it a bit clearer what its doing. Again, reference the scheduled job its running under in the logs